autokey-module/main.tf

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  default_suffix     = var.suffix != "" ? var.suffix : random_string.suffix.result
```

Comment:
If a suffix is provided, does it break the ability to terraform destroy and then apply back to back since the key name cannot be reused?

Reply:
Yes, if user explicit set the suffix (which is not the default behavior) it can break, depending on other configs.
For example, if `create_autokey_project` is `true`, even if suffix is set, it wouldn't break when destroying and re-applying because it would be the same key name but a different project.